src/Table/TableBody.js

```diff
@@ -139,7 +143,7 @@ class TableBody extends Component {
   }
 
   handleClickAway = () => {
-    if (this.props.deselectOnClickaway && this.state.selectedRows.length) {
+    if (this.props.deselectOnClickaway && (this.state.selectedRows.length || this.props.allRowsSelected)) {
```

Comment:
Since we clear the selectedRows when `allRowsSelected` is set, we need to check for that flag as well and deselect all if either is true.